### DISCUSSION TITLE
Statemanager to hold session?

### DISCUSSION THREAD

**Discussion by bengrau** [nextauthjs/next-auth#7055] [Help]
Hello,

I have a next.js project where handle a lot of user interactions. The app has many components where I require the user's session data to make various API calls. I could fetch the session data once in the top component and pass it down or get the session data in each component. However, I did a little bit of thinking: I was wondering if it would be good practice/idea to store the session data in the statemanager such as Zustand to avoid prop drilling and session refetching.

What do you guys think? Would that make sense or what is the best practice?

Thank you in advance!

**Comment by gavinthomas-valtech** [comment]
How about react query? I've started looking at this as i hadn't realised useSession doesn't dedupe request so means a lot of requests to /api/auth
 see RQ example here: https://github.com/nextauthjs/react-query/blob/main/index.js
